### PR TITLE
Fix RCFile optimized reader char decoding

### LINKED ISSUE #7734: hive.rcfile_optimized_reader_enabled=true CHAR regression
Steps:
`create table khaki_t WITH (format = 'RCTEXT') as select cast('khaki' as char(20)) khaki_c;`
`SELECT count(*) FROM khaki_t WHERE khaki_c = 'khaki[15spaceshere]';`

Current behavior:
```
_col0 
-------
     0 
```

Expected behavior:
```
_col0 
-------
     1
```

Workaround:
`set session hive.rcfile_optimized_reader_enabled=false;`

Regression was introduced in commits range:
`0c6c3b24ff37b4c35c228b68b64073eb58c03983..fdc4a384794c6d782d6c8e2e94a523109c552a27`

FYI: @dain

### PR DESCRIPTION
**Pull request by findepi**
Fixes #7734

### CODE REVIEW

**Comment by electrum** [comment]
Can you add a test for this?

**Comment by findepi** [comment]
Sure. In Hive queries, ok?

**Comment by dain** [comment]
Can you add a test to `AbstractTestRcFileReader`?

**Comment by dain** [comment]
Also if possible, AbstractTestHiveFileFormats, so we can catch this for other formats.

**Comment by findepi** [comment]
@dain `AbstractTestRcFileReader` looks suitable for round-trip testing but not suitable for testing run-time representation. If the run-time value is `' '`-padded, it will have no impact on its `String` representation.

I added a regression test in `TestHiveDistributedQueries` because this I knew how to do. 

I will need some guidance around `AbstractTestHiveFileFormats` / `AbstractTestRcFileReader` if i am to add something there.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java [hunk 1]
```diff
@@ -37,8 +37,8 @@
     private static final String PARQUET_OPTIMIZED_READER_ENABLED = "parquet_optimized_reader_enabled";
     private static final String MAX_SPLIT_SIZE = "max_split_size";
     private static final String MAX_INITIAL_SPLIT_SIZE = "max_initial_split_size";
-    private static final String RCFILE_OPTIMIZED_READER_ENABLED = "rcfile_optimized_reader_enabled";
-    private static final String RCFILE_OPTIMIZED_WRITER_ENABLED = "rcfile_optimized_writer_enabled";
+    public static final String RCFILE_OPTIMIZED_READER_ENABLED = "rcfile_optimized_reader_enabled";
+    public static final String RCFILE_OPTIMIZED_WRITER_ENABLED = "rcfile_optimized_writer_enabled";
```
Comment (pnowojski) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java]: move to the top

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java [hunk 2]
```diff
@@ -61,4 +64,30 @@ public void testOrderByChar()
...
+        testRcTextCharDecoding(true, false);
+        testRcTextCharDecoding(true, true);
+    }
+
+    private void testRcTextCharDecoding(boolean rcFileOptimizedWriterEnabled, boolean rcFileOptimizedReaderEnabled)
+            throws Exception
+    {
+        String catalog = getSession().getCatalog().get();
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(catalog, RCFILE_OPTIMIZED_WRITER_ENABLED, "" + rcFileOptimizedWriterEnabled)
```
Comment (fiedukow) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java]: Boolean.toString?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java [hunk 3]
```diff
@@ -61,4 +64,30 @@ public void testOrderByChar()
...
+                .setCatalogSessionProperty(catalog, RCFILE_OPTIMIZED_READER_ENABLED, "" + rcFileOptimizedReaderEnabled)
```
Comment (fiedukow) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java]: Boolean.toString?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java [hunk 4]
```diff
@@ -61,4 +64,30 @@ public void testOrderByChar()
 
         assertEquals(actual, expected);
     }
+
+    @Test
+    public void testRcTextCharDecoding()
+            throws Exception
+    {
+        testRcTextCharDecoding(false, false);
```
Comment (fiedukow) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java]: Having test method with to boolean parameter isn't too intutive, maybe we can use some annotation magic or something here?

Comment (findepi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java]: E.g.?

**Comment by dain** [comment]
Merged, thanks!

**Comment by findepi** [comment]
Thanks for review!

**Pull request closed without merging**